In [12]:
# 이미지 삭제

import os

src = r'한국인 피부상태 측정 데이터\데이터\Training\01.원천데이터\3. 스마트폰'

# 모든 하위 디렉터리 및 파일 탐색

for root, _, files in os.walk(src):
    
    for file in files:
        # 파일명이 ??_F.jpg 형식인지 확인
        if not file.endswith('_F.jpg'):
            # 조건에 맞지 않으면 파일 삭제
            file_path = os.path.join(root, file)
            os.remove(file_path)
            print(f"Deleted: {file_path}")

In [13]:
# 라벨링 데이터 삭제

import re

src = r"한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰"

pattern = re.compile(r'.+?_F_.+?\.json$')

for root, _, files in os.walk(src):
    
    for file in files:
        # 파일명이 ??_F.jpg 형식인지 확인
        if not pattern.match(file):
            # 조건에 맞지 않으면 파일 삭제
            file_path = os.path.join(root, file)
            os.remove(file_path)
            print(f"Deleted: {file_path}")

Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_00.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_01.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_02.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_03.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_04.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_05.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_06.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_07.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_L_08.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_R_00.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_R_01.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0006\0006_03_R_02.json
Deleted: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스

In [44]:
# folder_path = "한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰/0130"
# file_list = [f for f in os.listdir(folder_path) if f.endswith('.json')]  # 폴더 내 모든 .json 파일을 리스트로 만듬

In [35]:
# src = r"한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰"

# file_list = []

# for root, _, files in os.walk(src):

#     for f in files:
#         file_list.append(f)  # 폴더 내 모든 .json 파일을 리스트로 만듬



In [37]:
file_list

['0002_03_F_00.json',
 '0002_03_F_01.json',
 '0002_03_F_02.json',
 '0002_03_F_03.json',
 '0002_03_F_04.json',
 '0002_03_F_05.json',
 '0002_03_F_06.json',
 '0002_03_F_07.json',
 '0002_03_F_08.json',
 '0003_03_F_00.json',
 '0003_03_F_01.json',
 '0003_03_F_02.json',
 '0003_03_F_03.json',
 '0003_03_F_04.json',
 '0003_03_F_05.json',
 '0003_03_F_06.json',
 '0003_03_F_07.json',
 '0003_03_F_08.json',
 '0006_03_F_00.json',
 '0006_03_F_01.json',
 '0006_03_F_02.json',
 '0006_03_F_03.json',
 '0006_03_F_04.json',
 '0006_03_F_05.json',
 '0006_03_F_06.json',
 '0006_03_F_07.json',
 '0006_03_F_08.json',
 '0007_03_F_00.json',
 '0007_03_F_01.json',
 '0007_03_F_02.json',
 '0007_03_F_03.json',
 '0007_03_F_04.json',
 '0007_03_F_05.json',
 '0007_03_F_06.json',
 '0007_03_F_07.json',
 '0007_03_F_08.json',
 '0008_03_F_00.json',
 '0008_03_F_01.json',
 '0008_03_F_02.json',
 '0008_03_F_03.json',
 '0008_03_F_04.json',
 '0008_03_F_05.json',
 '0008_03_F_06.json',
 '0008_03_F_07.json',
 '0008_03_F_08.json',
 '0009_03_

In [40]:
import os
import json
import pandas as pd

In [47]:

# 작업 경로 설정
src = r"한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰"

# 데이터를 저장할 리스트
data = []

# 모든 하위 디렉터리 및 파일 탐색
for root, _, files in os.walk(src):
    for file_name in files:
        if file_name.endswith('.json'):  # JSON 파일만 처리
            file_path = os.path.join(root, file_name)
            
            # JSON 파일 읽기
            with open(file_path, 'r', encoding='utf-8') as f:
                json_data = json.load(f)
            
            # 필요한 정보만 추출
            info = json_data.get('info', {})
            images = json_data.get('images', {})
            annotations = json_data.get('annotations', {})
            equipment = json_data.get('equipment', {})

            # 'equipment'가 None일 경우 빈 딕셔너리로 처리
            if equipment is None:
                equipment = {}

            # 추출한 정보를 하나의 딕셔너리로 저장
            data.append({
                'filename': str(file_name).split('.')[0],
                'id': info.get('id'),
                'gender': info.get('gender'),
                'age': info.get('age'),
                'date': info.get('date'),
                'skin_type': info.get('skin_type'),
                'sensitive': info.get('sensitive'),
                'device': images.get('device'),
                'angle': images.get('angle'),
                'facepart': images.get('facepart'),
                'bbox': images.get('bbox'),
                'acne': annotations.get('acne'),
                'pigmentation_count': equipment.get('pigmentation_count', 0)  # 기본값 0
            })

# 리스트를 데이터프레임으로 변환
df = pd.DataFrame(data)

# 결과 출력
print(df.head())


       filename    id gender  age        date  skin_type  sensitive  device  \
0  0002_03_F_00  0002      F   50  2023-07-27          0          0       2   
1  0002_03_F_01  0002      F   50  2023-07-27          0          0       2   
2  0002_03_F_02  0002      F   50  2023-07-27          0          0       2   
3  0002_03_F_03  0002      F   50  2023-07-27          0          0       2   
4  0002_03_F_04  0002      F   50  2023-07-27          0          0       2   

   angle  facepart                      bbox  acne  pigmentation_count  
0      0         0        [0, 0, 1920, 2560]  None                 158  
1      0         1    [788, 691, 1510, 1005]  None                   0  
2      0         2  [1029, 1002, 1258, 1212]  None                   0  
3      0         3    [600, 1141, 727, 1410]  None                   0  
4      0         4  [1539, 1094, 1618, 1406]  None                   0  


In [59]:
want = df[df['filename'].str.endswith('00', na=False)]

In [60]:
want['bbox'].astype(str).unique()

array(['[0, 0, 1920, 2560]', '[0, 0, 360, 480]', '[0, 0, 600, 800]',
       '[0, 0, 768, 1024]', '[0, 0, 1919, 2560]', '[0, 0, 960, 1280]',
       '[0, 0, 720, 960]', '[0, 0, 2208, 2944]', '[0, 0, 480, 640]',
       '[0, 0, 962, 1280]', '[0, 0, 863, 1152]', '[0, 0, 2316, 3088]',
       '[0, 0, 2320, 3088]', '[0, 0, 1980, 2640]', '[0, 0, 2448, 3264]',
       '[0, 0, 2080, 2784]', '[0, 0, 979, 1306]', '[0, 0, 4248, 5664]',
       '[0, 0, 1956, 2608]', '[0, 0, 2556, 3408]', '[0, 0, 2004, 2672]',
       '[0, 0, 2544, 3392]', '[0, 0, 3880, 5184]', '[0, 0, 3456, 4608]',
       '[0, 0, 3024, 4032]', '[0, 0, 2388, 3184]', '[0, 0, 3000, 4000]',
       '[0, 0, 2736, 3648]', '[0, 0, 1104, 1472]', '[0, 0, 1080, 1440]',
       '[0, 0, 2472, 3296]', '[0, 0, 3488, 4640]', '[0, 0, 2460, 3280]',
       '[0, 0, 1392, 1856]', '[0, 0, 1932, 2576]', '[0, 0, 3492, 4656]',
       '[0, 0, 4896, 6528]', '[0, 0, 3184, 5664]'], dtype=object)

acne - 여드름 <br>
pigmentation_count - 색소 침착 <br>

width - 1920 <br>
height -  2560 <br>

sensitive - 0 : 아니오, 1 : 예 <br>

skin_type<br>
0 : 중성 <br>
1 : 건성 <br>
2 : 심한건성 <br>
3 : 복합건성 <br>
4 : 복합지성 <br>
5 : 지성


# 라벨링 데이터와 이미지 삭제 skin_type = 0(중성)

In [6]:

import os
import json
import shutil

src = "한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰"
other_dir = r'한국인 피부상태 측정 데이터\데이터\Training\01.원천데이터\3. 스마트폰'

for root, _, files in os.walk(src):
    want_file = []
    
    if files:
       want_file.append(files[0])
    else:
        print("hi")

    print(want_file)
    
    for file_name in want_file:
        
        if file_name.endswith('.json'):  # JSON 파일만 처리
            file_path = os.path.join(root, file_name)
            
            # JSON 파일 읽기
            with open(file_path, 'r', encoding='utf-8') as file:
                data = json.load(file)  # JSON 데이터를 Python 객체로 변환
            
            # skin_type이 0인 경우 처리
            if data.get("info", {}).get("skin_type") == 0:
                folder_to_delete = root  # JSON 파일이 포함된 폴더
                
                # 삭제 대상 폴더명 찾기
                target_folder_name = data["info"].get("id")  # 삭제 대상 폴더 이름
                
                # 현재 폴더 삭제
                shutil.rmtree(folder_to_delete)
                print(f"폴더 삭제: {folder_to_delete}")
                
                # # 다른 디렉토리에서 해당 이름의 폴더 삭제
                # for other_root, dirs, _ in os.walk(other_dir):
                #     for dir_name in dirs:
                #         if dir_name == target_folder_name:  # 폴더명이 동일한 경우
                #             target_folder_path = os.path.join(other_root, dir_name)
                #             shutil.rmtree(target_folder_path)  # 폴더 삭제
                #             print(f"폴더 삭제: {target_folder_path}")

                

hi
[]
['0002_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0002
['0003_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0003
['0006_03_F_00.json']
['0007_03_F_00.json']
['0008_03_F_00.json']
['0009_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0009
['0010_03_F_00.json']
['0011_03_F_00.json']
['0012_03_F_00.json']
['0014_03_F_00.json']
['0016_03_F_00.json']
['0017_03_F_00.json']
['0018_03_F_00.json']
['0019_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0019
['0020_03_F_00.json']
['0022_03_F_00.json']
['0023_03_F_00.json']
['0024_03_F_00.json']
['0025_03_F_00.json']
['0026_03_F_00.json']
['0027_03_F_00.json']
['0028_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0028
['0029_03_F_00.json']
['0030_03_F_00.json']
['0031_03_F_00.json']
['0032_03_F_00.json']
['0033_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰\0033
['0035_03_F_00.json']
['0036_03_F_00.json']
['

In [7]:
# 라벨링 데이터 이진 분류 1 = 건성, 0 = 지성

In [8]:
import os
import json
import shutil

src = r"한국인 피부상태 측정 데이터/데이터/Training/02.라벨링데이터/3. 스마트폰"

# 데이터를 저장할 리스트
data = []

# 모든 하위 디렉터리 및 파일 탐색
for root, _, files in os.walk(src):
    for file_name in files:

        if file_name.endswith('.json'):  # JSON 파일만 처리
            file_path = os.path.join(root, file_name)  # 전체 경로 생성
        
        with open(file_path, 'r', encoding='utf-8') as file:
            data = json.load(file)  # JSON 데이터를 Python 객체로 변환
        
         
        # 데이터 수정
        if "info" in data and "skin_type" in data["info"]:
            skin_type = data["info"]["skin_type"]
            if skin_type in [2, 3]:
                data["info"]["skin_type"] = 1
                print('건성으로')
            elif skin_type in [4, 5]:
                data["info"]["skin_type"] = 0
                print('지성으로')
        
        # 수정된 데이터를 JSON 파일로 저장
        with open(file_path, 'w', encoding='utf-8') as file:
            json.dump(data, file, ensure_ascii=False, indent=4)


건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로


0 = 지성 <br>
1 = 건성

# Validation 전처리

In [11]:
# 이미지 삭제

import os

src = r'한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰'

# 모든 하위 디렉터리 및 파일 탐색

for root, _, files in os.walk(src):
    
    for file in files:
        # 파일명이 ??_F.jpg 형식인지 확인
        if not file.endswith('_F.jpg'):
            # 조건에 맞지 않으면 파일 삭제
            file_path = os.path.join(root, file)
            os.remove(file_path)
            print(f"Deleted: {file_path}")

# 라벨링 데이터 삭제

import re

src = r"한국인 피부상태 측정 데이터/데이터/Validation/02.라벨링데이터/3. 스마트폰"

pattern = re.compile(r'.+?_F_.+?\.json$')

for root, _, files in os.walk(src):
    
    for file in files:
        # 파일명이 ??_F.jpg 형식인지 확인
        if not pattern.match(file):
            # 조건에 맞지 않으면 파일 삭제
            file_path = os.path.join(root, file)
            os.remove(file_path)
            print(f"Deleted: {file_path}")

Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0001\0001_03_L.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0001\0001_03_R.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0004\0004_03_L.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0004\0004_03_R.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0013\0013_03_L.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0013\0013_03_R.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0021\0021_03_L.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0021\0021_03_R.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0050\0050_03_L.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0050\0050_03_R.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0061\0061_03_L.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0061\0061_03_R.jpg
Deleted: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0072\0072_03_L.jpg
Deleted: 한국인

# Validation 이미지 라벨링 데이터 삭제 skin_type = 0 (중성)

In [9]:
import os
import json
import shutil

src = "한국인 피부상태 측정 데이터/데이터/Validation/02.라벨링데이터/3. 스마트폰"
other_dir = r'한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰'

for root, _, files in os.walk(src):
    want_file = []
    
    if files:
       want_file.append(files[0])
    else:
        print("hi")

    print(want_file)
    
    for file_name in want_file:
        
        if file_name.endswith('.json'):  # JSON 파일만 처리
            file_path = os.path.join(root, file_name)
            
            # JSON 파일 읽기
            with open(file_path, 'r', encoding='utf-8') as file:
                data = json.load(file)  # JSON 데이터를 Python 객체로 변환
            
            # skin_type이 0인 경우 처리
            if data.get("info", {}).get("skin_type") == 0:
                folder_to_delete = root  # JSON 파일이 포함된 폴더
                
                # 삭제 대상 폴더명 찾기
                target_folder_name = data["info"].get("id")  # 삭제 대상 폴더 이름
                
                # 현재 폴더 삭제
                shutil.rmtree(folder_to_delete)
                print(f"폴더 삭제: {folder_to_delete}")
                
                # 다른 디렉토리에서 해당 이름의 폴더 삭제
                for other_root, dirs, _ in os.walk(other_dir):
                    for dir_name in dirs:
                        if dir_name == target_folder_name:  # 폴더명이 동일한 경우
                            target_folder_path = os.path.join(other_root, dir_name)
                            shutil.rmtree(target_folder_path)  # 폴더 삭제
                            print(f"폴더 삭제: {target_folder_path}")

                

hi
[]
['0001_03_F_00.json']
['0004_03_F_00.json']
['0013_03_F_00.json']
['0021_03_F_00.json']
['0050_03_F_00.json']
['0061_03_F_00.json']
['0072_03_F_00.json']
['0082_03_F_00.json']
['0090_03_F_00.json']
['0111_03_F_00.json']
['0115_03_F_00.json']
['0117_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Validation/02.라벨링데이터/3. 스마트폰\0117
폴더 삭제: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0117
['0120_03_F_00.json']
['0144_03_F_00.json']
['0149_03_F_00.json']
['0159_03_F_00.json']
['0168_03_F_00.json']
['0186_03_F_00.json']
['0207_03_F_00.json']
['0249_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Validation/02.라벨링데이터/3. 스마트폰\0249
폴더 삭제: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0249
['0250_03_F_00.json']
['0257_03_F_00.json']
폴더 삭제: 한국인 피부상태 측정 데이터/데이터/Validation/02.라벨링데이터/3. 스마트폰\0257
폴더 삭제: 한국인 피부상태 측정 데이터\데이터\Validation\01.원천데이터\3. 스마트폰\0257
['0262_03_F_00.json']
['0275_03_F_00.json']
['0277_03_F_00.json']
['0307_03_F_00.json']
['0312_03_F_00.json']
['0321_03_F_00.json']
폴더 삭제: 한국인 피부상태

# Validation 데이터 이진 분류 1 = 건성, 0 = 지성

In [10]:
import os
import json

src = r"한국인 피부상태 측정 데이터/데이터/Validation/02.라벨링데이터/3. 스마트폰"

# 데이터를 저장할 리스트
data = []

# 모든 하위 디렉터리 및 파일 탐색
for root, _, files in os.walk(src):
    for file_name in files:

        if file_name.endswith('.json'):  # JSON 파일만 처리
            file_path = os.path.join(root, file_name)  # 전체 경로 생성
        
        with open(file_path, 'r', encoding='utf-8') as file:
            data = json.load(file)  # JSON 데이터를 Python 객체로 변환
        
         
        # 데이터 수정
        if "info" in data and "skin_type" in data["info"]:
            skin_type = data["info"]["skin_type"]
            if skin_type in [2, 3]:
                data["info"]["skin_type"] = 1
                print('건성으로')
            elif skin_type in [4, 5]:
                data["info"]["skin_type"] = 0
                print('지성으로')
        
        # 수정된 데이터를 JSON 파일로 저장
        with open(file_path, 'w', encoding='utf-8') as file:
            json.dump(data, file, ensure_ascii=False, indent=4)


건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
건성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
지성으로
